### DISCUSSION TITLE
WebXR not working with flutter WebView

### DISCUSSION THREAD

**Discussion by 0xhrsh** [google/model-viewer#3096] [Q&A]
hey there, I am opening model viewer inside a flutter android app, I am not getting the AR button when I set the AR-Modes to webxr. Does anyone know any workaround for that?

**Comment by elalish** [comment]
Indeed, WebXR is not  allowed in a WebView. However, it is enabled in the new and improved replacement: WebLayer.

**Reply by lupohirp** [reply]
Hi elalish, Sorry to bother you on such old answer.....but can you explain me what is WebLayer? i can't find any information on this one....

**Reply by elalish** [reply]
Hmm, actually it looks like it hasn't become a public API yet. Hopefully we can get support into WebView soon.
